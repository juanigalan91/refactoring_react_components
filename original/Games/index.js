import React from 'react';
import PropTypes from 'prop-types';

const namespace = 'games';
const Games = (props) => {
  const { games, title } = props;
  return (
    <div className={namespace}>
      <h1 key="title" className={`${namespace}__title`}>{title}</h1>
      <ul key={namespace} className={`${namespace}__list`}>
        {
          games.map((game) => {
            const baseClassName = 'game';

            return (
              <li className={baseClassName} key={game.id}>
                <a
                  href={game.link}
                  className={`${baseClassName}__link`}
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <img
                      src={game.src}
                      alt={game.title}
                      className={`${baseClassName}__img`}
                  />
                  <h3 className={`${baseClassName}__img_title`}>{game.title}</h3>
                </a>
              </li>
            );
          })
        }
      </ul>
    </div>
  );
};

Games.propTypes = {
  games: PropTypes.arrayOf(PropTypes.shape({
    link: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    src: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
  })).isRequired,
  title: PropTypes.string.isRequired
};

export default Games;
