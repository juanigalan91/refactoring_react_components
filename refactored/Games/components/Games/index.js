import React from 'react';
import PropTypes from 'prop-types';
import Game from '../Game';

const namespace = 'games';
const Games = (props) => {
  const { games, title } = props;
  return (
        <div className={namespace}>
            <h1 key="title" className={`${namespace}__title`}>{title}</h1>
            <ul key={namespace} className={`${namespace}__list`}>
                {
                    games.map(image => <li><Game {...image} key={image.id} /></li>)
                }
            </ul>
        </div>
  );
};

Games.propTypes = {
  games: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired
};

export default Games;
