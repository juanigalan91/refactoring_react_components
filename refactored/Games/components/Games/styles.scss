.images {
    background-color: #fff;
    border-radius: 4px;
    box-shadow: 0 1px 1px 0 rgba(0,0,0,.1);
    padding: 20px;
    margin-bottom: 20px;
    overflow-x: scroll; 

    &__list {
        display: flex;
        flex-direction: row;
        justify-content: space-evenly;
        text-align: center;
        margin: 0;
        padding: 0;
    }

    &__title {
        margin: 0 0 20px 0;
    }
}