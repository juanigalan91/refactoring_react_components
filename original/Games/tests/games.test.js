import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Games from '..';
import games from '../../../data';

describe('Games component', () => {
  it('renders the games component with the list of games', () => {
    const renderer = new ShallowRenderer();
    const component = (
            <Games games={games} title="Test title" />
    );
    expect(renderer.render(component)).toMatchSnapshot();
  });
});
