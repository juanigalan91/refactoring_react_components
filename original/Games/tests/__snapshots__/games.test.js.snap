// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Games component renders the games component with the list of games 1`] = `
<div
  className="games"
>
  <h1
    className="games__title"
  >
    Test title
  </h1>
  <ul
    className="games__list"
  >
    <li
      className="game"
    >
      <a
        className="game__link"
        href="https://www.amazon.com/Red-Dead-Redemption-2-PlayStation-4/dp/B01M5DZ525/ref=sr_1_3?keywords=ps4+games&qid=1551459125&s=gateway&sr=8-3"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Red Dead Redemption 2"
          className="game__img"
          src="https://m.media-amazon.com/images/I/81z80lFufgL._AC_UL872_FMwebp_QL65_.jpg"
        />
        <h3
          className="game__img_title"
        >
          Red Dead Redemption 2
        </h3>
      </a>
    </li>
    <li
      className="game"
    >
      <a
        className="game__link"
        href="https://www.amazon.com/Marvels-Spider-Man-PlayStation-4/dp/B01GW8YDLK/ref=sr_1_4?keywords=ps4+games&qid=1551459125&s=gateway&sr=8-4"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Marvels Spiderman"
          className="game__img"
          src="https://m.media-amazon.com/images/I/81d6JU6g1pL._AC_UL872_FMwebp_QL65_.jpg"
        />
        <h3
          className="game__img_title"
        >
          Marvels Spiderman
        </h3>
      </a>
    </li>
    <li
      className="game"
    >
      <a
        className="game__link"
        href="https://www.amazon.com/Marvels-Spider-Man-PlayStation-4/dp/B01GW8YDLK/ref=sr_1_4?keywords=ps4+games&qid=1551459125&s=gateway&sr=8-4"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="God of war"
          className="game__img"
          src="https://m.media-amazon.com/images/I/51po2bu7VnL._AC_UL872_FMwebp_QL65_.jpg"
        />
        <h3
          className="game__img_title"
        >
          God of war
        </h3>
      </a>
    </li>
    <li
      className="game"
    >
      <a
        className="game__link"
        href="https://www.amazon.com/Assassins-Creed-Odyssey-PlayStation-4-Standard/dp/B07DKYN13M/ref=sr_1_5?keywords=ps4+games&qid=1551460416&s=gateway&sr=8-5"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt="Assassins creed Odyssey"
          className="game__img"
          src="https://m.media-amazon.com/images/I/81da48KsFpL._AC_UL872_FMwebp_QL65_.jpg"
        />
        <h3
          className="game__img_title"
        >
          Assassins creed Odyssey
        </h3>
      </a>
    </li>
  </ul>
</div>
`;
